src/v/cluster/types.h

```diff
@@ -1631,21 +1642,23 @@ struct property_update<tristate<T>>
 struct incremental_topic_updates
   : serde::envelope<
       incremental_topic_updates,
-      serde::version<4>,
+      serde::version<5>,
       serde::compat_version<0>> {
     static constexpr int8_t version_with_data_policy = -1;
     static constexpr int8_t version_with_shadow_indexing = -3;
     static constexpr int8_t version_with_batch_max_bytes_and_local_retention
       = -4;
     static constexpr int8_t version_with_segment_ms = -5;
     static constexpr int8_t version_with_schema_id_validation = -6;
+    static constexpr int8_t version_with_initial_retention = -7;
     // negative version indicating different format:
     // -1 - topic_updates with data_policy
     // -2 - topic_updates without data_policy
     // -3 - topic_updates with shadow_indexing
     // -4 - topic update with batch_max_bytes and retention.local.target
     // -6 - topic updates with schema id validation
-    static constexpr int8_t version = version_with_schema_id_validation;
+    // -7 - topic updates with initial retention
+    static constexpr int8_t version = version_with_initial_retention;
```

Comment:
i think we do not longer need ADL serialization here

Reply:
Created an issue to remove all ADL's `version_` fields there. Having it half-done is confusing https://github.com/redpanda-data/redpanda/issues/14257